### PR TITLE
Enhance test_put_along_axis, test_take_along_axis.

### PR DESCRIPTION
**Pull request by robinw0928**

### CODE REVIEW

**Comment by manopapad** [comment]
Can you merge from latest branch-22.12, and trigger the CI again? The OpenMP failure may be because of an issue that Irina fixed with https://github.com/nv-legate/cunumeric/pull/665.

File: tests/integration/test_put_along_axis.py [hunk 1]
```diff
@@ -54,6 +67,191 @@ def test_ndim(ndim):
...
+            np_values = mk_seq_array(np, s)
+            num_values = mk_seq_array(num, s)
+            np_a = np_arr.copy()
+            num_a = num_arr.copy()
+            np.put_along_axis(np_a, np_indices, np_values, axis=axis)
+            num.put_along_axis(num_a, num_indices, num_values, axis=axis)
+            assert np.array_equal(np_a, num_a)
+
+
+def test_empty_indice():
```
Comment (ipdemes) [inline tests/integration/test_put_along_axis.py:110]: did you miss `s` in `indices`?

File: tests/integration/test_put_along_axis.py [hunk 2]
```diff
@@ -54,6 +67,191 @@ def test_ndim(ndim):
...
 
 
+def test_full():
+    shape = (3, 4, 5)
+    ndim = len(shape)
+    np_arr = mk_seq_array(np, shape)
+    num_arr = mk_seq_array(num, shape)
+
+    for axis in range(-1, ndim):
+        size = shape[axis]
```
Comment (ipdemes) [inline tests/integration/test_put_along_axis.py:77]: can this for loop be moved to pytest's parametrize like:
`@pytest.mark.parametrize("ndim", range(-1, 3))`

**Comment by ipdemes** [comment]
I left a couple of minor comments, it looks good otherwise. 
I will create a separate PR that fixes all issues found in the test and remove `xfail`s

**Comment by ipdemes** [comment]
Please update you branch with branch-22.12 and feel free to merge it when CI passes

**Review by ipdemes — APPROVED** [review]

File: tests/integration/test_put_along_axis.py (hunk 1 above)
Comment (robinw0928) [reply tests/integration/test_put_along_axis.py:111]: No. In test_values, `indices` has fixed shape (3, 4, 5) while the shape of `values` varies.

File: tests/integration/test_put_along_axis.py (hunk 2 above)
Comment (robinw0928) [reply tests/integration/test_put_along_axis.py]: Fixed.

**Comment by ipdemes** [comment]
@robinw0928 : feel free to merge this PR

**Pull request merged**